http/peer-id-auth.md

```diff
@@ -0,0 +1,137 @@
+# Peer ID Authentication over HTTP
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+| --------------- | ------------- | ------ | --------------- |
+| 1A              | Working Draft | Active | r0, 2023-01-23  |
+
+Authors: [@MarcoPolo]
+
+[@MarcoPolo]: https://github.com/MarcoPolo
+
+Interest Group: Same as [HTTP](README.md)
+
+## Introduction
+
+This spec defines an authentication scheme of libp2p Peer IDs in accordance with
+[RFC 9110](https://datatracker.ietf.org/doc/html/rfc9110). The authentication
+scheme is called `Libp2p-PeerID`.
+
+## Mutual Client and Server Peer ID Authentication
+
+1. The server initiates the authentication by responding to a request that must
+   be authenticated with the response header `WWW-Authenticate: Libp2p-PeerID
+   challenge-client="<base64-encoded-challenge>`. The challenge MUST be
+   indistinguishable from random data. The Server MAY randomly generate this
+   data, or MAY use an server-encrypted value. If using random data the
+   server SHOULD store the challenge temporarily until the authentication is
+   done. The challenge SHOULD be at least 32 bytes.
+
+1. The client sends a request and sets the `Authorization`
+   [header](https://www.rfc-editor.org/rfc/rfc9110.html#section-11.6.2) header
+   to the following:
+   ```
+   Libp2p-PeerID peer-id="<encoded-peer-id-bytes>",[challenge-server="<base64-encoded-challenge-server>",]sig="<base64-signature-bytes>"]
+   ```
+
+   * The `challenge-server` parameter is optional. The client should set it if
+     the client wants to authenticate the server.
+   * The peer-id is encoded per the string encoding described in the [peer-ids spec](../peer-ids/peer-ids.md).
+   * The signature is over the concatenated result of:
+   ```
+     <varint-length> + "origin=" + server-name + 
+     [<varint-length> + "challenge-server=" + base64-encoded-client-chosen-challenge-server + ]
+     <varint-length> + "challenge-client=" + base64-encoded-challenge
+   ```
+   * Strings are UTF-8 encoded.
+   * If the challenge server was omitted in the `Authorization` header it MUST
+     be omitted in the signature.
+   * If provided, the client-chosen `challenge-server` MUST be randomly generated.
+   * The client-chosen `challenge-server` SHOULD be at least 32 bytes.
+   * The client MUST use the same server-name as what is used for the TLS
+     session.
+   * If the client _only_ wants to authenticate the server and the server does
```

Comment:
Can we make this a separate section? 
Mutual Authentication
Client Only Authentication
Server Only Authentication

Reply:
That makes sense. Maybe an overview section and then separate sections for specifics.